### PR TITLE
Filters component 

### PR DESCRIPTION
**Pull request by georgemataca**
## Description
Please enter a short description of this PR.

## Type of change

_Please delete options that are not relevant._

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:
- [ ] I have made corresponding changes to the documentation
- [x] My changes do not generate any new warnings

~- [ ] I have written tests or updated existing tests for this change~ *update when testing harness has been added*

~- [ ] New and existing tests pass locally~ *update when testing harness has been added*

### CODE REVIEW

File: src/ui/Filters/Filters.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+            {title}
+          </Title>
+          <ClearButton
+            transparent
+            color="terracota"
+            variant="primary"
+            uppercase={false}
+            onClick={this._clearFilters}
+            spacing="0"
+            width={["70px", null,"200px", null]}
```
Comment (georgemataca) [inline src/ui/Filters/Filters.js]: Need suggestion on how to overwrite default styles using media queries for the button so I don't fall back to using the responsive method.

File: src/ui/Filters/FilterOption.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { Spring, animated } from 'react-spring';
+import Icon from 'ui/Icon/Icon';
+
+const Container = styled.div`
+  border-bottom: 0.5px solid ${props => props.theme.colors.terracota.main};
```
Comment (rfarine) [inline src/ui/Filters/FilterOption.js]: For _all_ pixel values in here, please use the `rem()` helper method from `polished`

File: src/ui/Filters/FilterOption.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+  line-height: 20px;
+  margin: 0 10px;
+  margin-top: 1px;
+  padding-top: 1px;
+  text-align: center;
+  vertical-align: middle;
+  width: 20px;
+  color: ${props => props.theme.colors.white.main};
+  font-weight: 700;
+  font-size: ${12/16}rem;
```
Comment (rfarine) [inline src/ui/Filters/FilterOption.js]: This will just be `rem('12px')`

File: src/ui/Filters/FilterOption.js [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+  margin-top 4px;
+  font-size: ${12/16}rem;
+  font-weight: 700;
+  font-family: ${props => props.theme.text.secondary};
+  text-transform: uppercase;
+  letter-spacing: 1.85px;
+  flex: 1
+`;
+
+class FilterOption extends React.Component {
```
Comment (rfarine) [inline src/ui/Filters/FilterOption.js]: Please use `PureComponent`

File: src/ui/Filters/FilterOption.js [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+  static propTypes = {
+    isOpen: PropTypes.bool,
+    name: PropTypes.string.isRequired,
+    data: PropTypes.arrayOf(PropTypes.shape({})),
+    active: PropTypes.arrayOf(PropTypes.string),
+    onHeaderClick: PropTypes.func,
+    onChange: PropTypes.func,
+  };
```
Comment (rfarine) [inline src/ui/Filters/FilterOption.js]: PropTypes are declared outside of most all of our components. Let's stick to that style choice.

Comment (rfarine) [reply src/ui/Filters/FilterOption.js]: ie) at bottom of file before default export:
```
FilterOption.propTypes = { ... };
```

File: src/ui/Filters/FilterOption.js [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+    const { isOpen, name, active, onHeaderClick } = this.props;
+
+    return (
+      <Container>
+        <Header onClick={onHeaderClick}>
+          <FilterName
+          >
+            {name}
+          </FilterName>{' '}
+          {active.length ? (
```
Comment (rfarine) [inline src/ui/Filters/FilterOption.js]: Don't need to do this ternary stuff anymore. Remember to just do:

`{active.length && <Counter>{active.length}</Counter>}`

File: src/ui/Filters/FilterOption.js [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+
+  _handleChange = option => {
+    this.props.onChange(
+      this.props.active.includes(option)
+        ? this.props.active.filter(item => item !== option)
+        : [...this.props.active, option]
+    );
+  };
+
+  renderOptions = style => {
```
Comment (rfarine) [inline src/ui/Filters/FilterOption.js]: I wouldn't make this its own function. It's easier to read if it's just declared within the spring component in the render() method.

File: src/ui/Filters/Filters.js [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+  text-align: right;
+`;
+  
+class Filters extends React.Component {
+  
+  state = {
+    openFilter: null,
+  };
+    
+  static propTypes = {
```
Comment (rfarine) [inline src/ui/Filters/Filters.js]: Same comment as above about propTypes

File: src/ui/Filters/Filters.js [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+  @media ${props => props.theme.queries.desktop} {
+    width: 200px;
+  }
+`;
+
+const ClearButton = styled(Button)`
+  text-align: right;
+`;
+  
+class Filters extends React.Component {
```
Comment (rfarine) [inline src/ui/Filters/Filters.js]: Same comment as above about PureComponent

File: src/ui/Filters/Filters.js [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+  static propTypes = {
+    title: PropTypes.string.isRequired,
+    options: PropTypes.arrayOf(PropTypes.shape({}).isRequired).isRequired,
+    activeFilters: PropTypes.shape({}).isRequired,
+    clearFilters: PropTypes.func.isRequired,
+    setFilter: PropTypes.func.isRequired,
+    children: PropTypes.node.isRequired,
+  };
+
+  static defaultProps = {
```
Comment (rfarine) [inline src/ui/Filters/Filters.js]: Also same comment as propTypes, but `Filters.defaultProps = {}`

File: src/ui/Filters/FilterOption.js [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+
+  static defaultProps = {
+    isOpen: false,
+    data: [],
+    active: [],
+    onHeaderClick: null,
+    onChange: null,
+  };
+
+  _handleChange = option => {
```
Comment (rfarine) [inline src/ui/Filters/FilterOption.js]: Let's rename _all_ functions that start with these underscores. The underscores are unnecessary.

File: src/ui/Filters/Filters.js [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+    activeFilters: {},
+  };
+
+  _handleFilterTouch = (filter) => {
```
Comment (rfarine) [inline src/ui/Filters/Filters.js]: remove underscore

File: src/ui/Filters/Filters.js [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+
+          if (!data) {
+            return <div>Filter with section: {accessor} not found.</div>;
+          }
+
+          return React.cloneElement(child, {
+            active,
+            data,
+            isOpen: this.state.openFilter === accessor && !this.state.forceCollapse,
+            onHeaderClick: () => {
```
Comment (rfarine) [inline src/ui/Filters/Filters.js]: `onHeaderClick` and `onChange` are going to generate a new function on every render. This is bad for performance.. https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md

Maybe try to find a way to refactor this?

File: src/ui/Filters/Filters.stories.js [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeFilters: initialFilters,
+      filterOptions: filterOptions,
+    }
+  }
+
+  clearFilters = () => this.setState({ activeFilters: initialFilters});
+  setFilter = ({ accessor, filter }) => this.setState({ activeFilters: { ...this.state.activeFilters, [accessor]: filter }});
```
Comment (rfarine) [inline src/ui/Filters/Filters.stories.js]: We don't want to be creating a brand new component here in our stories... Stories should just accept stubbed out state/props and only render the component we're testing. Check out other stories to see how we simulate setting state (ie. Drawer.stories.js)

File: src/ui/Filters/Filters.js (hunk 13 above)
Comment (georgemataca) [reply src/ui/Filters/Filters.js]: Out of ideas for the moment

File: src/ui/Filters/Filters.js [hunk 15]
```diff
@@ -0,0 +1,207 @@
...
+          });
+        })}
+      </FiltersContainer>
+    );
+  }
+}
+
+Filters.propTypes = {
+  title: PropTypes.string.isRequired,
+  options: PropTypes.arrayOf(PropTypes.shape({}).isRequired).isRequired,
```
Comment (rfarine) [inline src/ui/Filters/Filters.js]: Let's define really clearly what shape we expect

File: src/ui/Filters/Filters.js [hunk 16]
```diff
@@ -0,0 +1,207 @@
...
+              onClick={this.handleClearFilters}
+              spacing="0"
+              height={rem('20px')}
+              weight="normal"
+              >
+              Clear All
+            </ClearButton>
+          </ButtonContainer>
+        </Header>
+        {React.Children.map(children, (child) => {
```
Comment (rfarine) [inline src/ui/Filters/Filters.js]: Why not just map over an array of objects to create the `<Option>`s?

ie) 
```
<Filters options={[ { accessor: "hi", isOpen: false, onChange: this.onChange } ]} />

....

{this.props.options.map(option => <Option accessor={option.accessor} isOpen={option.isOpen} ... />)}
```

**Pull request merged**